BlockEntity/BEFruitpress.cs

```diff
@@ -390,28 +385,35 @@ private bool InteractMashContainer(IPlayer byPlayer, BlockSelection blockSel)
                 int removeItems;
                 if (hprops.LitresPerItem == null)
                 {
-                    var availableLitres = (float)handStack.Attributes.GetDecimal("juiceableLitresLeft");
-                    transferableLitres = (float)Math.Min(availableLitres, juiceableLitresCapacity - juiceableLitresLeft);
-                    // the juiceableLitresLeft is per item if we have a stack of multiple
-                    // so we only remove one item and add the liters for one
-                    removeItems = 1;
+                    // the juiceableLitresLeft and juiceableLitresTransfered are per item if we have a stack of multiple
+                    // so we check if the press is full and only remove and add the liters for one item at a time
+                    if (juiceableLitresLeft + juiceableLitresTransfered + transferableLitres + usedLitres > juiceableLitresCapacity)
+                    {
+                        (Api as ICoreClientAPI)?.TriggerIngameError(this, "fullcontainer", Lang.Get("Container is full, press out juice and remove the mash before adding more"));
+                        return false;
+                    }
+
+                    if (juiceableLitresLeft + juiceableLitresTransfered <= 0) mashStack.Attributes.SetDouble("squeezeRel", handStack.Attributes.GetDouble("squeezeRel", 1)); //Sets the mash in the press to be as squeezed as the one you're adding by hand.
                 } else
-                {  
-                    float desiredTransferSizeLitres = byPlayer.Entity.Controls.ShiftKey ? (float)hprops.LitresPerItem : Math.Min(handStack.StackSize, 4) * (float)hprops.LitresPerItem;
-                    transferableLitres = (float)Math.Min(desiredTransferSizeLitres, juiceableLitresCapacity - juiceableLitresLeft);
+                {
+                    int desiredTransferAmount = Math.Min(handStack.StackSize, byPlayer.Entity.Controls.ShiftKey ? 1 : byPlayer.Entity.Controls.CtrlKey ? 32 : 4); //Fixed it so you can actually shift click just 1 in, but also added a ctrl click 32 option just to speed things along.
+
+                    while (desiredTransferAmount * (float)hprops.LitresPerItem + juiceableLitresLeft + juiceableLitresTransfered > juiceableLitresCapacity) desiredTransferAmount -= 1; //Quick calculation of the total number of fruit you can add without going over the total capacity to make sure it's never adding more than a single fruit worth of juice at a time.
 
-                    removeItems = (int)(transferableLitres / hprops.LitresPerItem);
+                    transferableLitres = desiredTransferAmount * (float)hprops.LitresPerItem;
+                    removeItems = desiredTransferAmount;
                 }
 
-                if (transferableLitres > 0) {
+                if (removeItems > 0) { //Changed to only run this when it is going to remove items instead of just any time it thinks there's juice to add.

```

Comment:
Changed to only run this when it is going to remove items instead of just any time it thinks there's juice to add.